tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/OptimizerCore.scala

```diff
@@ -111,6 +111,9 @@ private[optimizer] abstract class OptimizerCore(
           disableOptimisticOptimizations = true
           transformIsolatedBody(Some(myself), thisType, params, resultType, body)
       }
+      val newBody =
+        if (name.name == "init___") tryElimStoreModule(newBody1)
```

Comment:
Oh it would *work*, but would do useless work. Since we're after `StoreModule`s, this is only relevant to the constructor odd module classes, and those are always called `init___`. It's an early filter.